src/main/kotlin/uk/gov/justice/digital/hmpps/sentenceplan/services/GoalService.kt

```diff
@@ -87,11 +89,19 @@ class GoalService(
     return relatedAreasOfNeedList
   }
 
+  /**
+   * This function expects a full Goal object - it cannot be used for updating individual fields in
+   * a Goal object identified by the UUID unless the rest of the goal object is passed in.
+   * In particular the GoalEntity.merge function will remove all Related Areas of Need from a GoalEntity
```

Comment:
can we add the word "intentionally" before "remove" here please?

Reply:
Added in https://github.com/ministryofjustice/hmpps-sentence-plan/pull/145/commits/a8ed9522ef8cab14ffb98a1a61f0474928502505